enhancements/cloud-integration/ibm/service-endpoint-dynamic-override.md

```diff
@@ -0,0 +1,126 @@
+
+### Workflow Description
+
+**cluster administrator** is a human user responsible managing an existing openshift custer deployed on IBM infrastructure.
+
+1. The cluster administrator wishes to use private IBM Cloud endpoints.
+2. The cluster administrator identifies the services that they wish to update (ie IAM and resource controller) and identifies the endpoints for these services.
+3. The cluster administrator updates the infrastructure object to contain a list of overrides where each element is the name of the service and the endpoint to use for that service. `oc edit infrastructure  -n default cluster`
+4. Once the service endpoint override update has been processed/reconciled, components can act on the change (if applicable) and use in future operations (note: may need to be restarted to pick up the change).
+
+**cccmo** is an operator responsible for watching updates to the infrastructure object and perforning updates once any value(s) are set. 
+
+1. The cccmo reconciliation loop observes that the IBMCloudPlatform spec within the infrastructure object has been set,
+2. The cccmo validates the endpoints and then updates the IBMCloudPlatformStatus and cloud config.
+
+
+### API Extensions
+
+* Extend `IBMCloudPlatformSpec` to contain service endpoint field that users may define as desired overrides.
+
+```
+type IBMCloudPlatformSpec struct {
+	// serviceEndpoints is a list of custom endpoints which will override the default
+	// service endpoints of an IBM Cloud service. These endpoints are consumed by
+	// components within the cluster to reach the respective IBM Cloud Services.
+	// +listType=map
+	// +listMapKey=name
+	// +optional
+	ServiceEndpoints []IBMCloudServiceEndpoint `json:"serviceEndpoints,omitempty"`
+}
+```
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+n/a
+
+#### Standalone Clusters
+
+n/a
+
+#### Single-node Deployments or MicroShift
+
+n/a
+
+### Implementation Details/Notes/Constraints
+
+* API will be updated such that [IBMCloudPlatformSpec](https://github.com/openshift/api/blob/4c27e61e5554ea8506947d019770e5a04c3c4a36/config/v1/types_infrastructure.go#L1522) will have a field for `IBMCloudServiceEndpoints` similar to the existing field in [IBMCloudPlatformSpec](https://github.com/openshift/api/blob/4c27e61e5554ea8506947d019770e5a04c3c4a36/config/v1/types_infrastructure.go#L1549)
+* CCCMO will be updated so that config sync controller via the IBM `CloudConfigTransformer` reads in endpoint settings within the spec of the infrastructure object and updates the corresponding infrastructure status and cloud config to reflect those set values.
+
+### Risks and Mitigations
+
+Users may cause service interruptions for their cluster should they define invalid overrides. This is mitigated by performing validation on the endpoint as there requirements are understood at time of implementation for IBM Cloud. 
```

Comment:
I'd like to see how much of this can be done at API time, vs controller time. Earlier validation is better generally